sdk/scheduler/src/main/java/com/mesosphere/sdk/metrics/Metrics.java

```diff
@@ -114,4 +121,32 @@ public static void record(Protos.TaskStatus taskStatus) {
         final String metricName = String.format("task_status.%s", taskStatus.getState().name().toLowerCase());
         metrics.counter(metricName).inc();
     }
+
+    public static void setPlanStatus(Optional<String> namespace, String planName, Status status) {
+        final String metricName = namespace.isPresent() ?
+                String.format("plan_status.%s.%s", namespace.get(), planName)
+                :
+                String.format("plan_status.%s", planName);
+
+        Map<String, Gauge> gauges = metrics.getGauges((name, metric) -> name.equals(metricName));
+        if (gauges.size() > 0) {
+            ((PlanGauge) gauges.get(metricName)).setStatus(status);
+        } else {
+            PlanGauge gauge = new PlanGauge();
+            metrics.gauge(metricName, () -> gauge);
+        }
+    }
+
+    static class PlanGauge implements Gauge<Integer> {
+        private Status status;
+
+        public void setStatus(Status status) {
+            this.status = status;
+        }
+
+        @Override
+        public Integer getValue() {
+            return status.ordinal();
```

Comment:
Rather than just getting the ordinal, I think it'd make sense to just specify the values here via e.g. a switch statement. We also don't need to expose the full granularity of plan statuses. More code, but less brittle. Could be something like:
```
ERROR = -1 # ERROR or unknown status
COMPLETE = 0 # COMPLETE
IN_PROGRESS = 1 # IN_PROGRESS, STARTING, STARTED, PREPARED
PENDING = 2 # PENDING or WAITING
```
Not sure if those are the best numbers to use, but those 4 states seem like enough.